code/game/machinery/bounty_board.dm

```diff
@@ -0,0 +1,223 @@
+		var/obj/item/card/id/current_card = I
+		if(current_card.registered_account)
+			current_user = current_card.registered_account
+			return TRUE
+		to_chat(user, "There's no account assigned with this ID.")
+		return TRUE
+	if(istype(I, /obj/item/bounty_card))
+		if(!current_user)
+			playsound(src, 'sound/machines/buzz-sigh.ogg', 20, TRUE)
+			return TRUE
+		var/obj/item/bounty_card/curr_bounty = I
+		var/datum/station_request/curr_request = curr_bounty.new_request
+		if(!curr_request)
+			playsound(src, 'sound/machines/buzz-sigh.ogg', 20, TRUE)
+			return TRUE
+		curr_request.req_number = current_user.account_id
+		curr_request.owner_account = current_user
+		GLOB.request_list += list(curr_request)
+		request_number++
+		for(var/obj/i in GLOB.allbountyboards)
+			i.say("New bounty has been added!")
+			playsound(i.loc, 'sound/effects/cashregister.ogg', 30, TRUE)
+		qdel(I)
+	if(I.tool_behaviour == TOOL_WRENCH)
+		to_chat(user, "<span class='notice'>You start [anchored ? "un" : ""]securing [name]...</span>")
+		I.play_tool_sound(src)
+		if(I.use_tool(src, user, 30))
+			playsound(loc, 'sound/items/deconstruct.ogg', 50, TRUE)
+			if(machine_stat & BROKEN)
+				to_chat(user, "<span class='warning'>The broken remains of [src] fall on the ground.</span>")
+				new /obj/item/stack/sheet/metal(loc, 3)
+				new /obj/item/shard(loc)
+			else
+				to_chat(user, "<span class='notice'>You [anchored ? "un" : ""]secure [name].</span>")
+				new /obj/item/wallframe/bounty_board(loc)
+			qdel(src)
+
+/obj/machinery/bounty_board/ui_interact(mob/user, ui_key, datum/tgui/ui, force_open, datum/tgui/master_ui, datum/ui_state/state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "RequestKiosk", name, ui_x, ui_y, master_ui, state)
+		ui.open()
+
+/obj/machinery/bounty_board/ui_data(mob/user)
+	var/list/data = list()
+	var/list/formatted_requests = list()
+	var/list/formatted_applicants = list()
+	for(var/i in GLOB.request_list)
+		if(!i)
+			continue
+		var/datum/station_request/request = i
+		formatted_requests += list(list("owner" = request.owner, "value" = request.value, "description" = request.description, "acc_number" = request.req_number))
+		if(request.applicants)
+			for(var/datum/bank_account/j in request.applicants)
+				formatted_applicants += list(list("name" = j.account_holder, "request_id" = request.owner_account.account_id, "requestee_id" = j.account_id))
+	var/obj/item/card/id/id_card = user.get_idcard()
+	if(id_card?.registered_account)
+		current_user = id_card.registered_account
+	if(current_user)
+		data["AccountName"] = current_user.account_holder
+	data["Requests"] = formatted_requests
+	data["Applicants"] = formatted_applicants
+	return data
+
+/obj/machinery/bounty_board/ui_act(action, list/params)
+	if(..())
+		return
+	var/current_ref_num = params["request"]
+	var/current_app_num = params["applicant"]
+	var/datum/bank_account/request_target
+	if(GLOB.request_list.len)
```

Comment:
Not what i meant. If params["request"] is empty you shouldn't be iterating the list.